src/routes/safe/components/Transactions/TxList/TxInfoTransfer.tsx

```diff
@@ -14,21 +15,27 @@ export const TxInfoTransfer = ({ txInfo }: { txInfo: Transfer }): ReactElement |
   const assetInfo = useAssetInfo(txInfo)
   const [details, setDetails] = useState<Details | undefined>()
 
+  const makeTitle = (txDirection: string, amountWithSymbol: string) => (
+    <Text size="xl" as="span">
+      {txDirection === TransferDirection.INCOMING ? 'Received ' : 'Sent '}
+      <Text size="xl" as="span" strong>
+        {amountWithSymbol}
+      </Text>
+      {txDirection === TransferDirection.INCOMING ? ' from:' : ' to:'}
+    </Text>
+  )
+
   useEffect(() => {
     if (assetInfo && assetInfo.type === 'Transfer') {
-      if (txInfo.direction.toUpperCase() === 'INCOMING') {
-        setDetails({
-          title: `Received ${assetInfo.amountWithSymbol} from:`,
-          address: txInfo.sender.value,
-          name: txInfo.sender.name || undefined,
-        })
-      } else {
-        setDetails({
-          title: `Send ${assetInfo.amountWithSymbol} to:`,
-          address: txInfo.recipient.value,
-          name: txInfo.recipient.name || undefined,
-        })
-      }
+      const txDirection = txInfo.direction.toUpperCase()
+      setDetails({
+        title: makeTitle(txDirection, assetInfo.amountWithSymbol),
+        address: txDirection === TransferDirection.INCOMING ? txInfo.sender.value : txInfo.recipient.value,
+        name:
+          txDirection === TransferDirection.INCOMING
+            ? txInfo.sender.name || undefined
```

Comment:
You can remove `undefined` here.

Reply:
The `name` is `string | null` => **Type 'string | null' is not assignable to type 'string | undefined'**
I had to propagate the type change down the component hierarchy

Reply:
I see. If you put the ternary in brackets then you only need the one fallback though.